server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader9to10.java

```diff
@@ -165,8 +165,10 @@ private void validateACLs(ServerContext context) {
 
           if (((path.equals(Constants.ZROOT) || path.equals(Constants.ZROOT + Constants.ZINSTANCES))
               && !acls.equals(ZooDefs.Ids.OPEN_ACL_UNSAFE))
-              || (!privateWithAuth.equals(acls) && !publicWithAuth.equals(acls))) {
-            log.error("ZNode at {} has unexpected ACL: {}", path, acls);
+              || (!acls.containsAll(privateWithAuth) && !acls.containsAll(publicWithAuth))) {
```

Comment:
I'm looking at adding an option in zoo-info-viewer that can dump the acls in zookeeper  that could be used before running the upgrade.

I was leaning towards checking that our processes have cdrw permissions - either with the accumulo digest or via wide-open, world-anyone - that ensures that we have the permissions necessary to run but does not provide any security assurance.  

Otherwise, other than a few nodes like /accumulo and /accumulo/instance I think all nodes under /accumulo/[IID]/tables could be protected with the accumulo digest and not allow any other zookeeper user id access - but that could impact any uses that have external utilities that access ZooKeeper (say to read FATE status) that might not have the accumulo digest set - they probably should, but they might have been okay with world-read.

Reply:
IIRC from ZooKeeperInitializer:

ZROOT -> ZooDefs.Ids.OPEN_ACL_UNSAFE
ZROOT + ZINSTANCES -> ZooDefs.Ids.OPEN_ACL_UNSAFE
ZROOT/IID/config -> ZooUtil.PRIVATE
Everything else has ZooUtil.PUBLIC

Reply:
Upon second thought, I think it's probably not a good idea to check set equality, as I said earlier. Users may have customized their ACLs after installation, and that should be allowed. We only need to check that the minimum permissions we need, that the Accumulo system user has full permissions on the nodes for this instance.